src/engine/ruleset/decoders/fortinet/fortimail.yml

```diff
@@ -0,0 +1,199 @@
+    # date=2017-1-6 time=07:11:41 device_id=uaUten log_id=nby log_part=mve type=event subtype=config pri=low user=isau ui=rautodi(10.96.97.81) module=pis submodule=nsequat msg=doloreme
+    - event.original: >-
+        date=<~log.date> time=<~log.time> device_id=<observer.serial_number> log_id=<event.code> log_part=<~log.rsa.misc.reference_id1>
+        type=<event.action> subtype=<~log.rsa.internal.messageid> pri=<log.level> user=<user.name> ui=<~log.rsa.network.network_service>\(<source.ip>\) module=<~log.fill> submodule=<~log.fill> msg=<~log.rsa.internal.event_desc>
+    # date=2018-3-25 time=09:31:24 device_id=obeataev log_id=umf log_part=olesti type=event subtype=config pri=low user=quaeabil ui=emip module=aturQu submodule=itesse msg="iamqui"
+    - event.original: >-
+        date=<~log.date> time=<~log.time> device_id=<observer.serial_number> log_id=<event.code> log_part=<~log.rsa.misc.reference_id1>
+        type=<event.action> subtype=<~log.rsa.internal.messageid> pri=<log.level> user=<user.name> ui=<~log.rsa.network.network_service> module=<~log.fill> submodule=<~log.fill> msg="<~log.rsa.internal.event_desc>"
+    # date=2016-6-5 time=21:33:08 device_id=ntutl log_id=caecatc log_part=onsequat type=event subtype=update pri=low msg="edquiano"
+    - event.original: >-
+        date=<~log.date> time=<~log.time> device_id=<observer.serial_number> log_id=<event.code> log_part=<~log.rsa.misc.reference_id1>
+        type=<event.action> subtype=<~log.rsa.internal.messageid> pri=<log.level> msg="<~log.rsa.internal.event_desc>"
+
+normalize:
+  - map:
+    - event.dataset: fortinet.fortimail
+    - event.module": fortinet
+    - fileset.name": fortimail
+    - input.type": log
+    - observer.type: Firewall
+    - observer.vendor: Fortinet
+    - observer.product: FortiMail
+    - rsa.misc.reference_id1: $~log.rsa.misc.reference_id1
+    - rsa.misc.reference_id: $event.code
+    - rsa.misc.msgIdPart1: $event.action
+    - rsa.misc.msgIdPart2: $~log.rsa.internal.messageid
+    - rsa.internal.messageid: $event.action
+    - rsa.internal.messageid: $~log.rsa.internal.messageid
+    - rsa.misc.event_type: $event.action
+    - rsa.misc.severity: $log.level
+    - rsa.misc.filter: $~log.rsa.misc.filter
+    - rsa.misc.result: $~log.rsa.misc.result
+    - rsa.network.domain: $server.domain
+    - rsa.misc.log_session_id: $~log.rsa.misc.filter
+    - rsa.email.email_src: $~log.rsa.email.email_src
+    - rsa.misc.client: $~log.rsa.misc.client
+    - rsa.web.fqdn: $~log.rsa.web.fqdn
+    - rsa.misc.context: $~log.rsa.misc.context
+    - rsa.email.email_dst: $~log.rsa.email.email_dst
+    - rsa.misc.virusname: $~log.rsa.misc.virusname
+    - rsa.misc.disposition: $~log.rsa.misc.disposition
+    - rsa.misc.filter: $~log.rsa.misc.filter
+    - rsa.email.subject: $~log.rsa.email.subject
+    - rsa.email.subject: $~log.rsa.misc.event_type
+    - rsa.internal.event_desc: $~log.rsa.internal.event_desc
+    - rsa.misc.checksum: $~log.rsa.misc.checksum
```

Comment:
Please sort alphabetically if possible making sure expressions are still valid in case subsequent calls have dependencies .

On VSCode you could easily use this extension Tyriar.sort-lines

Reply:
Done!